File: programmers/튜플/64065.java
import java.util.*;
import java.util.stream.Collectors;

// 프로그래머스 튜플
class Solution {
    public int[] solution(String s) {
        String[] strArray = s.substring(1, s.length() - 1).split("");

        List<String> tupleList = new ArrayList<>();

        String tuple = "";
        for (String str : strArray) {
            tuple += str;

            if (str.equals("}")) {
                if (tuple.charAt(0) == ',') {
                    tupleList.add(tuple.substring(1));

                } else {
                    tupleList.add(tuple);
                }

                tuple = "";
            }
        }

        tupleList = tupleList.stream()
                .sorted(Comparator.comparing(String::length))
                .collect(Collectors.toList());

        Map<String, Boolean> map = new HashMap<>();
        List<Integer> result = new ArrayList<>();
        for (String str : tupleList) {
            String[] tempArray = str.substring(1, str.length() - 1).split(",");

            for (String temp : tempArray) {
                if (map.get(temp) == null) {
                    map.put(temp, true);

                    result.add(Integer.parseInt(temp));
                }
            }

        }

        return result.stream().mapToInt(i -> i).toArray();
    }
}

File: programmers/튜플/64065.kt
class Solution {
    fun solution(s: String): IntArray {
        val charArray: CharArray = s.toCharArray();

        var tupleList: MutableList<String> = ArrayList();

        var tuple = "";

        for (i in 1 until charArray.size - 1) {
            tuple = tuple.plus(charArray[i]);

            if (charArray[i] == '}') {
                if (tuple.toCharArray()[0] == ',') {
                    tupleList.add(tuple.substring(1));

                } else {
                    tupleList.add(tuple);
                }

                tuple = ""
            }
        }

        tupleList = tupleList.sortedBy { it.length }.toMutableList();

        val map: MutableMap<String, Boolean> = HashMap();
        var result: MutableList<Int> = ArrayList();

        for (str in tupleList) {
            val tempArray: List<String> = str.substring(1, str.length - 1).split(",");

            for (temp in tempArray) {
                if (map[temp] == null) {
                    map[temp] = true;

                    result.add(temp.toInt());
                }
            }
        }

        return result.toIntArray();
    }
}